### PR TITLE
Add magnitude offset code and fix numerous small bugs.

### PR DESCRIPTION
**Pull request by erykoff**
This PR makes a whole bunch of changes ...

* Better selection from catalogs, limiting magnitude and signal-to-noise (avoiding bad faint stars)
* A second-stage "magnitude offset" (very physically unmotivated) is there to correct PS1 at the brighter end by comparing to XP
* Improved plotting ranges
* Warnings suppressed where necessary
* Filtering of output catalogs, so that stars that are outside the magnitude/color/signal-to-noise range will not have values in the transformed catalog
* Related, any object that is bad in all bands in the transformed catalog will not be output
* PS1 duplicates are removed
* Matching is reversed so that each Gaia star can only have one match from the other catalogs.
* A null color transformation for DES->DES has also been included.

Note that the XP to DES transformation was run and put into colorterm directory prior to using XP as a basis for the PS1 magnitude corrections.

### CODE REVIEW

File: python/lsst/the/monster/utils.py [hunk 1]
```diff
@@ -27,6 +27,10 @@ def read_stars(path, indices, allow_missing=False):
     """
     stars = None
     for index in indices:
+        fname = os.path.join(path, str(index) + ".fits")
+        if not os.path.isfile(fname):
+            if allow_missing:
+                continue
         try:
             temp = SimpleCatalog.readFits(os.path.join(path, str(index) + ".fits"))
```
Comment (jeffcarlin) [inline python/lsst/the/monster/utils.py:35]: Should be `temp = SimpleCatalog.readfits(fname)` ?

File: tests/test_splinecolorterms.py [hunk 2]
```diff
@@ -273,6 +327,73 @@ def test_spline_overwrite(self):
             spline.save(filename, overwrite=True)
 
 
+class MonsterMagSpineFitterTest(lsst.utils.tests.TestCase):
```
Comment (jeffcarlin) [inline tests/test_splinecolorterms.py:330]: "SpineFitter" --> "SplineFitter"?

**Comment by jeffcarlin** [comment]
> * Filtering of output catalogs, so that stars that are outside the magnitude/color/signal-to-noise range will not have values in the transformed catalog

For clarification -- are objects only removed when they have NaN fluxes in all bands? (I think that's what I'm seeing, but wanted to make sure.) Or is this a super-conservative "keep only stars within the fitting/SNR ranges" approach? If the latter, I am confused because I thought we wanted to bring all stars forward, and just set their transformed fluxes to NaN if they're outside the fitting ranges.

**Review by jeffcarlin — APPROVED** [review]
All looks good (a couple of minor comments to address).

**Comment by erykoff** [comment]
@jeffcarlin So on this PR I changed it so that any star that is matched but has no valid transformation fluxes is culled from the output catalog.  I thought this would make downstream processing a bit smoother/faster and the transformed files a bit smaller.  However, if you think it would be better to output all the matched sources whether or not they have valid transformations I could revert this commit.

**Comment by jeffcarlin** [comment]
> @jeffcarlin So on this PR I changed it so that any star that is matched but has no valid transformation fluxes is culled from the output catalog. I thought this would make downstream processing a bit smoother/faster and the transformed files a bit smaller. However, if you think it would be better to output all the matched sources whether or not they have valid transformations I could revert this commit.

I don't have strong opinions about it, and can't think of compelling cases for keeping them around, so I think it's fine.

**Pull request merged**